internal/util/util_test.go

```diff
@@ -200,10 +198,10 @@ func Test_SplitMap(t *testing.T) {
 				},
 				"name": "John",
 				"projects": []msa{
-					{"name": "Project2", "other": []msa{{"y": "y2"}}, "status": "completed", "vals": []any{1, 2, 3}},
+					{"name": "Project2", "other": []msa{{}, {"y": "y2"}}, "status": "completed", "vals": []any{1, 2, 3}},
 					{"name": "Project3", "other": []msa{{"y": "y3"}}},
-					{"name": "Project4"},
-					{"name": "Project1", "other": []msa{{"y": "y1"}}, "status": "in progress", "vals": []any{4, 5, 6, 7}},
+					{"name": "Project4", "other": []msa{{}}},
+					{"name": "Project1", "other": []msa{{}, {"y": "y1"}, {}, {}}, "status": "in progress", "vals": []any{4, 5, 6, 7}},
```

Comment:
At first, I thought the `SplitMap()` function should not return these empty maps in an array like that, but it's actually better that they do - it essentially maintains all of the array entries in here, which enables the caller to correctly get the diff between the 2 arrays. (if some entries were missing, the diff would not be correct)